src/rust/engine/src/intrinsics/dep_inference.rs

```diff
@@ -109,6 +111,66 @@ impl PreparedInferenceRequest {
     }
 }
 
+#[pyfunction]
+fn parse_dockerfile_info(deps_request: Value) -> PyGeneratorResponseNativeCall {
+    PyGeneratorResponseNativeCall::new(async move {
+        let context = task_get_context();
+
+        let core = &context.core;
+        let store = core.store();
+        let prepared_inference_request = PreparedInferenceRequest::prepare(
+            deps_request,
+            &store,
+            "Dockerfile",
+            dockerfile::IMPL_HASH,
+        )
+        .await?;
+        in_workunit!(
+            "parse_dockerfile_info",
+            Level::Debug,
+            desc = Some(format!(
+                "Determine Dockerfile info for {:?}",
+                &prepared_inference_request.inner.input_file_path
+            )),
+            |_workunit| async move {
+                let result: ParsedDockerfileDependencies = get_or_create_inferred_dependencies(
+                    core,
+                    &store,
+                    prepared_inference_request,
+                    |content, request| {
+                        dockerfile::get_info(content, request.inner.input_file_path.into())
+                    },
+                )
+                .await?;
+
+                let result = Python::with_gil(|py| {
+                    externs::unsafe_call(
+                        py,
+                        core.types.parsed_dockerfile_info_result,
+                        &[
+                            result.path.to_object(py).into(),
+                            result.build_args.to_object(py).into(),
+                            result.copy_source_paths.to_object(py).into(),
+                            result.copy_build_args.to_object(py).into(),
+                            result.from_image_build_args.to_object(py).into(),
+                            result
```

Comment:
Another example of something wholly unnecessary but which is done to match the old parser.